### PR TITLE
fix: code smells suggested by sonar

### PR DESCRIPTION
**Pull request by jayanta2018**
Handles 20 code smells and most of them are switching from field injection to
constructor injection

### CODE REVIEW

File: src/main/java/uk/nhs/tis/sync/job/CommonSyncJobTemplate.java [hunk 1]
```diff
@@ -31,11 +31,16 @@ public abstract class CommonSyncJobTemplate<T> implements RunnableJob {
...
 
-  @Autowired
-  private EntityManagerFactory entityManagerFactory;
+  protected final ApplicationEventPublisher applicationEventPublisher;
+
+  private final EntityManagerFactory entityManagerFactory;
 
-  @Autowired(required = false)
-  protected ApplicationEventPublisher applicationEventPublisher;
+  @Autowired
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/tis/sync/job/CommonSyncJobTemplate.java:38]: I ran the app to check.  As it's the only constructor, we don't need to annotate them.

File: src/main/java/uk/nhs/tis/sync/job/PersonPlacementTrainingBodyTrustJob.java [hunk 2]
```diff
@@ -27,19 +27,24 @@
...
 
-  @Autowired
-  private EntityManagerFactory entityManagerFactory;
-  @Autowired
   private PersonRepository personRepository;
-  @Autowired
+
   private SqlQuerySupplier sqlQuerySupplier;
 
+  @Autowired
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/tis/sync/job/PersonPlacementTrainingBodyTrustJob.java:39]: We don't need to annotate a sole constructor.

File: src/main/java/uk/nhs/tis/sync/job/PostEmployingBodyTrustJob.java [hunk 3]
```diff
@@ -34,16 +33,21 @@
...
-  private static final Logger LOG = LoggerFactory.getLogger(PostEmployingBodyTrustJob.class);
-  @Autowired
   private PostTrustRepository postTrustRepository;
-  @Autowired
-  private EntityManagerFactory entityManagerFactory;
-  @Autowired
+
   private SqlQuerySupplier sqlQuerySupplier;
 
+  public PostEmployingBodyTrustJob(EntityManagerFactory entityManagerFactory,
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/tis/sync/job/PostEmployingBodyTrustJob.java:43]: :smile:

File: src/main/java/uk/nhs/tis/sync/job/reval/RevalCurrentPlacementSyncJob.java [hunk 4]
```diff
@@ -25,8 +28,11 @@ public class RevalCurrentPlacementSyncJob extends RevalPersonChangedJobTemplate
           + " AND (dateFrom = ':today' OR dateTo = ':yesterday')"
           + " ORDER BY traineeId LIMIT :pageSize";
 
-  public RevalCurrentPlacementSyncJob(RabbitMqTcsRevalTraineeUpdatePublisher rabbitMqPublisher) {
-    super(rabbitMqPublisher);
+  @Autowired
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/tis/sync/job/reval/RevalCurrentPlacementSyncJob.java:31]: We don't need to annotate a sole constructor.

File: src/main/java/uk/nhs/tis/sync/job/reval/RevalCurrentPmSyncJob.java [hunk 5]
```diff
@@ -25,8 +28,11 @@ public class RevalCurrentPmSyncJob extends RevalPersonChangedJobTemplate {
           + " AND (programmeEndDate = ':endDate' OR programmeStartDate = ':startDate')"
           + " ORDER BY personId LIMIT :pageSize";
 
-  public RevalCurrentPmSyncJob(RabbitMqTcsRevalTraineeUpdatePublisher rabbitMqPublisher) {
-    super(rabbitMqPublisher);
+  @Autowired
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/tis/sync/job/reval/RevalCurrentPmSyncJob.java:31]: We don't need to annotate a sole constructor.

File: src/main/java/uk/nhs/tis/sync/job/reval/RevalPersonChangedJobTemplate.java [hunk 6]
```diff
@@ -4,18 +4,30 @@
...
+ * This is a template abstract class for Revalidation Person job change
+ * <p>
+ * Its purpose is to publish Persons based on their current Placement into Rabbit message queue
+ */
 public abstract class RevalPersonChangedJobTemplate extends
     PersonDateChangeCaptureSyncJobTemplate<Long> {
 
   private final RabbitMqTcsRevalTraineeUpdatePublisher rabbitMqPublisher;
 
+  @Autowired
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/tis/sync/job/reval/RevalPersonChangedJobTemplate.java:25]: We don't need to annotate a sole constructor.

**Comment by hihilary** [comment]
Looks good to me.

**Review by hihilary — APPROVED** [review]

**Pull request merged**